kotsadm/pkg/store/s3pg/reporting_store.go

```diff
@@ -126,3 +151,89 @@ func (s S3PGStore) getK8sVersion() (string, error) {
 	}
 	return k8sVersion.GitVersion, nil
 }
+
+func getAdminIDConfigMap() (*corev1.ConfigMap, error) {
+
+	cfg, err := k8sconfig.GetConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get kubernetes config")
+	}
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get clientset")
+	}
+	namespace := os.Getenv("POD_NAMESPACE")
+	existingConfigmap, err := clientset.CoreV1().ConfigMaps(namespace).Get(context.TODO(), configMapName, metav1.GetOptions{})
+	if err != nil && !kuberneteserrors.IsNotFound(err) {
+		return nil, errors.Wrap(err, "failed to get configmap")
+	} else if kuberneteserrors.IsNotFound(err) {
+		return nil, nil
+	}
+	if existingConfigmap != nil {
+		logger.Infof("Existing config map %v", existingConfigmap.Data["id"])
+		return existingConfigmap, nil
+	}
+	return nil, nil
+
+}
+
+// CreateAdminIDConfigMap creates an id for an kotsadm instance and stores in configmap
+func CreateAdminIDConfigMap(clusterID string) (*corev1.ConfigMap, error) {
+	cfg, err := k8sconfig.GetConfig()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get kubernetes config")
+	}
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get clientset")
+	}
+	configmap := corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "ConfigMap",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      configMapName,
+			Namespace: os.Getenv("POD_NAMESPACE"),
+			Labels: map[string]string{
+				"kots.io/kotsadm": "true",
```

Comment:
`kotstypes.KotsadmKey:  KotsadmLabelValue` in `github.com/replicatedhq/kots/pkg/kotsadm/types` package

Reply:
another suggestion, just use
```
Labels: kotsadmtypes.GetKotsadmLabels(),
```